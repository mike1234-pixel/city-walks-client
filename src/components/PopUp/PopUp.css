/** @format */

.popup-container-cookies {
  position: relative;
  z-index: 20;
}

.popup-cookies {
  background: rgba(63, 63, 63, 0.842);
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  color: white;
  font-size: 30px;
  padding: 20px;
  display: flex;
  flex-direction: column;
}
