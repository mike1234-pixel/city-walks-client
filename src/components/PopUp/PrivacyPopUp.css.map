{"version":3,"sourceRoot":"","sources":["PrivacyPopUp.scss"],"names":[],"mappings":"AAAA;AAEA;EACE;EACA;;;AAGF;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;;AAGF;EACE;;;AAGF;EACE;;;AAGF;AAAA;AAAA;EAGE;;;AAGF;AAAA;AAAA;EAGE","file":"PrivacyPopUp.css"}