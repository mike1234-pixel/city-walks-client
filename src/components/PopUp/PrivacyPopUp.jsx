import { MDBBtn, MDBContainer } from "mdbreact"
import { Link } from "react-router-dom"
import { connect } from 'react-redux'
import "./PrivacyPopUp.scss"

const PrivacyPopUp = (props) => {

    const { setPrivacyPopupVisible } = props

    const handlePrivacyPopup = () => {
        setPrivacyPopupVisible(false)
        localStorage.setItem("popupVisible", false)
    }

    return (
        <MDBContainer>
            <div className="popup-container-cookies">
                <div className="popup-cookies">
                    <p>To make City Walks work, we log user data. By using City Walks, you agree to our <Link to="/privacy" className="popup-link">Privacy Policy</Link>, including cookie policy.</p>
                    <p className="popup-google-note">This site is protected by reCAPTCHA and the <a target="_blank" href="https://policies.google.com/privacy" className="popup-link">Google Privacy Policy</a> and <a target="_blank" href="https://policies.google.com/terms" className="popup-link">Terms of Service</a> apply.</p>
                    <MDBBtn onClick={handlePrivacyPopup} className="popup-accept-btn">Accept</MDBBtn>
                </div>
            </div>
        </MDBContainer>
    )
}

const mapDispatchToProps = (dispatch) => {
    return {
        setPrivacyPopupVisible: (boolValue) => dispatch({ type: 'SET_POPUP_VISIBLE', boolValue }),
    }
}

export default connect(null, mapDispatchToProps)(PrivacyPopUp);