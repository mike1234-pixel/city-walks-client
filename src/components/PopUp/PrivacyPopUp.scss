/** @format */

.popup-container-cookies {
  position: relative;
  z-index: 20;
}

.popup-cookies {
  background: rgba(63, 63, 63, 0.842);
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  color: white;
  font-size: 30px;
  padding: 20px;
  display: flex;
  flex-direction: column;
}

.popup-google-note {
  font-size: 15px;
}

.popup-link {
  color: var(--blue);
}

.popup-link:hover,
.popup-link:focus,
.popup-link:active {
  color: #2bbbad;
}

.popup-accept-btn:hover,
.popup-accept-btn:active,
.popup-accept-btn:focus {
  background: var(--blue) !important;
}
