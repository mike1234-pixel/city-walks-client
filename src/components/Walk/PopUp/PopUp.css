/** @format */

.popup-container {
  position: relative;
}

.popup {
  background: rgba(63, 63, 63, 0.842);
  position: fixed;
  top: 0;
  left: 0;
  z-index: 5;
  width: 100vw;
  height: 100vh;
  display: flex;
}

.popup-img {
  width: 80vw;
  height: 80vh;
  margin: auto;
  border: 2px black solid;
}

@media (min-width: 1000px) {
  .popup-img {
    width: 40vw;
  }
}

/* sample map img */
/* https://coding-video.s3.eu-west-2.amazonaws.com/mapimg.JPG */
