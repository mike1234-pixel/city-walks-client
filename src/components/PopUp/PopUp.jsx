import { useContext } from "react"
import { LoginContext } from "../../context/LoginContext"
import { MDBBtn } from "mdbreact"
import { Link } from "react-router-dom"
import "./PopUp.css"

const PopUp = () => {

    const { handlePopup } = useContext(LoginContext)

    return (
        <div className="popup-container-cookies">
            <div className="popup-cookies">
                <p>To make City Walks work, we log user data. By using City Walks, you agree to our <Link to="/privacy">Privacy Policy</Link>, including cookie policy.</p>
                <MDBBtn onClick={handlePopup}>Accept</MDBBtn>
            </div>
        </div>
    )
}

export default PopUp