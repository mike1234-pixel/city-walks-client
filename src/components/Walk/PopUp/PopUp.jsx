import './PopUp.css'

const PopUp = (props) => {

    const { mapImg } = props

    return (
        <div className="popup-container">
            <div className="popup">
                {/* orientation will be portrait */}
                <img className="popup-img" src={mapImg}/>      
            </div>
        </div>
    )
}

export default PopUp