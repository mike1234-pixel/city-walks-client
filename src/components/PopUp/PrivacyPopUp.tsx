import React from "react";
import { MDBBtn, MDBContainer } from "mdbreact";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { Action, bindActionCreators, Dispatch } from "redux";
import { setPrivacyPopupVisible } from "../../actions/actions";
import * as Actions from "../../types/Actions";
import "./PrivacyPopUp.scss";

interface Props {
  setPrivacyPopupVisible: (popupVisible: boolean) => Actions.SetPopupVisible;
}

const PrivacyPopUp: React.FC<Props> = (props: Props) => {
  const { setPrivacyPopupVisible } = props;

  const handlePrivacyPopup: () => void = () => {
    setPrivacyPopupVisible(false);
    localStorage.setItem("popupVisible", "false");
  };

  return (
    <MDBContainer>
      <div className="popup-container-cookies">
        <div className="popup-cookies">
          <p>
            To make City Walks work, we log user data. By using City Walks, you
            agree to our{" "}
            <Link to="/privacy" className="popup-link">
              Privacy Policy
            </Link>
            , including cookie policy.
          </p>
          <p className="popup-google-note">
            This site is protected by reCAPTCHA and the{" "}
            <a
              target="_blank"
              href="https://policies.google.com/privacy"
              className="popup-link"
            >
              Google Privacy Policy
            </a>{" "}
            and{" "}
            <a
              target="_blank"
              href="https://policies.google.com/terms"
              className="popup-link"
            >
              Terms of Service
            </a>{" "}
            apply.
          </p>
          <MDBBtn onClick={handlePrivacyPopup} className="popup-accept-btn">
            Accept
          </MDBBtn>
        </div>
      </div>
    </MDBContainer>
  );
};

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    dispatch,
    ...bindActionCreators({ setPrivacyPopupVisible }, dispatch),
  };
};

export default connect(null, mapDispatchToProps)(PrivacyPopUp);
