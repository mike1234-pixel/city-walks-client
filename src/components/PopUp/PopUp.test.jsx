import renderer from 'react-test-renderer'
import { LoginContextProvider } from "../../context/LoginContext"
import { BrowserRouter } from 'react-router-dom';
import PopUp from './PopUp'

// snapshot test
it('PopUp renders correctly', () => {  

  const tree = renderer
    .create(
        <LoginContextProvider>
            <BrowserRouter>
                <PopUp/>
            </BrowserRouter>
        </LoginContextProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});