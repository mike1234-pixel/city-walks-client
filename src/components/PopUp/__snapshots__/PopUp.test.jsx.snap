// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopUp renders correctly 1`] = `
<div
  className="popup-container-cookies"
>
  <div
    className="popup-cookies"
  >
    <p>
      To make City Walks work, we log user data. By using City Walks, you agree to our 
      <a
        href="/privacy"
        onClick={[Function]}
      >
        Privacy Policy
      </a>
      , including cookie policy.
    </p>
    <button
      className="btn-default btn Ripple-parent"
      data-test="button"
      onClick={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      Accept
      <div
        className="Ripple "
        data-test="waves"
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        }
      />
    </button>
  </div>
</div>
`;
